itis.playground/Pages/homework.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,523 @@
+    }
+    
+    
+    var numerator: Int
+    var denominator: Int
+    
+    init(_ numerator: Int, over denominator: Int) {
+        self.numerator = numerator
+        self.denominator = denominator
+    }
+    
+    convenience init() {
+        self.init(0,over: 1)
+    }
+    
+    static prefix func -(fraction: Fraction) -> Fraction {
+        return Fraction(fraction.numerator * -1, over: fraction.denominator)
+    }
+    
+    // sum
+    static func +(lhs: Fraction, rhs: Fraction) -> Fraction {
+        lhs.reduce()
+        rhs.reduce()
+        
+        let newNumerator = lhs.numerator * rhs.denominator + lhs.denominator * rhs.numerator
+        let newDenominator = lhs.denominator * rhs.denominator
+        let f = Fraction(newNumerator, over: newDenominator)
+        f.reduce()
+        return f
+        
+    }
+    
+    // difference
+    static func -(lhs: Fraction, rhs: Fraction) -> Fraction {
+        return lhs + (-rhs);
+    }
+    
+    // product
+    static func *(lhs: Fraction, rhs: Fraction) -> Fraction {
+        let f = Fraction(lhs.numerator * rhs.numerator,
+                         over: lhs.denominator * rhs.denominator)
+        f.reduce()
+        return f
+    }
+    
+    // inverse
+    func inverse() -> Fraction {
+        let f = Fraction(denominator, over: numerator)
+        f.reduce()
+        return f
+    }
+    
+    // division
+    static func /(lhs: Fraction, rhs: Fraction) -> Fraction {
+        let f = lhs * rhs.inverse()
+        f.reduce()
+        return f
+    }
+    
+    // divides both the numerator and the denominator by
+    // their gcd
+    func reduce() {
+       var negativeCache = false
+        if numerator < 0 {
+            negativeCache = !negativeCache
+            numerator *= -1
+        }
+        if denominator < 0 {
+            negativeCache = !negativeCache
+            denominator *= -1
+        }
+        
+        let divisor = gcd(numerator,denominator)
+        numerator /= divisor
+        denominator /= divisor
+        
+        if negativeCache {
+            numerator *= -1
+        }
+    }
+    
+    private func gcd(_ num1: Int, _ num2: Int) -> Int {
+        let remainder = num1 % num2
+        
+        return remainder == 0 ? num2 : gcd(num2,remainder);
+    }
+    
+    private func lcm(_ num1: Int, _ num2: Int) -> Int {
+        return (num1 * num2) / gcd(num1,num2)
+    }
+}
+
+// MARK: TEST for TASK 2
+let fraction = Fraction(2, over: 3)
+let fraction2 = Fraction(1, over: 4)
+
+let sum = fraction + fraction2
+print(sum) // sum = 11/12
+
+let diff = fraction - fraction2
+print(diff) // diff = 5/12
+
+let product = fraction * fraction2
+print(product) // product = 1/6
+
+let quotient = fraction / fraction2
+print(quotient) // quotient = 8/3
+
+
+// MARK: - TASK 3
+
+// Дано:
+protocol Student {
+    var health: Int { get set }
+    var damage: Int { get set }
+    var chant: String? { get set }
+    var name: String { get set }
```

Comment:
```suggestion
    let name: String { get set }
```
имя скорее всего у тебя не будет меняться